src/apps/sturdy/fantom/sturdy.lending.token-fetcher.ts

```diff
@@ -0,0 +1,95 @@
+import { Inject } from '@nestjs/common';
+import axios from 'axios';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { ZERO_ADDRESS } from '~app-toolkit/constants/address';
+import { Register } from '~app-toolkit/decorators';
+import { ContractType } from '~position/contract.interface';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { AppTokenPosition } from '~position/position.interface';
+import { BaseToken } from '~position/token.interface';
+import { Network } from '~types/network.interface';
+
+import { SturdyContractFactory } from '../contracts';
+import { STURDY_DEFINITION } from '../sturdy.definition';
+
+const appId = STURDY_DEFINITION.id;
+const groupId = STURDY_DEFINITION.groups.lending.id;
+const network = Network.FANTOM_OPERA_MAINNET;
+
+type VaultMonitoringResponse = {
+  chain: string;
+  tokens: string;
+  decimals: number;
+  address: string;
+  supply: number;
+  price: number;
+  base: number;
+  reward: number;
+  rewards: {
+    CRV: number;
+    CVX: number;
+  };
+  url: number;
+  tvl: number;
+  active: boolean;
+}[];
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class FantomSturdyLendingTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(SturdyContractFactory) private readonly sturdyContractFactory: SturdyContractFactory,
+  ) { }
+
+  async getPositions() {
+    const multicall = this.appToolkit.getMulticall(network);
+    const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+    const ethToken = baseTokens.find(t => t.address === ZERO_ADDRESS);
+    if (!ethToken) return [];
+
+    const endpoint = 'https://us-central1-stu-dashboard-a0ba2.cloudfunctions.net/getVaultMonitoring';
+    const tokenData = await axios.get<VaultMonitoringResponse>(endpoint).then(v => v.data);
+
+    const tokens = tokenData.map(async (data) => {
+      const symbol = data.tokens;
+      const underlyingTokens: BaseToken[] = [];
+
+      let contractFactoryFunc;
+      if (symbol === 'USDC') contractFactoryFunc = this.sturdyContractFactory.sturdyUsdc;
+      else if (symbol === 'fUSDT') contractFactoryFunc = this.sturdyContractFactory.sturdyFusdt;
+      else if (symbol === 'DAI') contractFactoryFunc = this.sturdyContractFactory.sturdyDai;
```

Comment:
The contract ABIs are actually the same for all three, so you can replace `sturdy-usdc.json`, `sturdy-fusdt.json` and `sturdy-dai.json` with one single `sturdy-token.json`.

Re-run the contract fetcher generator with `pnpm studio generate:contract-factory sturdy`, then use `this.sturdyContractFactory.sturdyToken({ address: data.address, network })` in all cases.